src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -0,0 +1,1244 @@
+        }
+
+        jexls = VariantContextUtils.initializeMatchExps(selectNames, selectExpressions);
+
+        // Prepare the sample names and types to be used by the corresponding filters
+        samples = createSampleNameInclusionList(vcfRods);
+        selectedTypes = createSampleTypeInclusionList();
+
+        // Look at the parameters to decide which analysis to perform
+        discordanceOnly = discordanceTrack != null;
+        if (discordanceOnly) {
+            logger.info("Selecting only variants discordant with the track: " + discordanceTrack.getName());
+        }
+
+        concordanceOnly = concordanceTrack != null;
+        if (concordanceOnly) {
+            logger.info("Selecting only variants concordant with the track: " + concordanceTrack.getName());
+        }
+
+        if (mendelianViolations) {
+            sampleDB = initializeSampleDB();
+            mv = new MendelianViolation(mendelianViolationQualThreshold, false, true);
+        }
+
+        selectRandomFraction = fractionRandom > 0;
+        if (selectRandomFraction) {
+            logger.info("Selecting approximately " + 100.0*fractionRandom + "% of the variants at random from the variant track");
+        }
+
+        // Prepare the variant IDs to keep and to be removed for use by the IDs filter
+        IDsToKeep = getIDsFromFile(rsIDFile);
+        IDsToRemove = getIDsFromFile(XLrsIDFile);
+
+        VariantContextWriterBuilder vcWriterBuilder =
+                new VariantContextWriterBuilder()
+                    .setOutputFile(outFile)
+                    .setOutputFileType(VariantContextWriterBuilder.OutputType.VCF)
+                    .unsetOption(Options.INDEX_ON_THE_FLY);
+        if (lenientVCFProcesssing) {
+            vcWriterBuilder = vcWriterBuilder.setOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER);
+        }
+        vcfWriter = vcWriterBuilder.build();
+
+        final File refFile = referenceArguments.getReferenceFile();
+        final SAMSequenceDictionary samDictionary = this.getBestAvailableSequenceDictionary();
+        Set<VCFHeaderLine> actualLines;
+        if (refFile != null && samDictionary!= null) {
+            actualLines = withUpdatedContigsAsLines(headerLines, refFile, samDictionary, suppressReferenceName);
+        }
+        else {
+            actualLines = headerLines;
+        }
+        vcfWriter.writeHeader(new VCFHeader(actualLines, samples));
+    }
+
+    @Override
+    public void apply(VariantContext vc, ReadsContext readsContext, ReferenceContext ref, FeatureContext featureContext) {
+
+        if (mendelianViolations && invertLogic((mv.countFamilyViolations(sampleDB, samples, vc) == 0), invertMendelianViolations)) {
+            return;
+        }
+
+        if (discordanceOnly && !isDiscordant(vc, featureContext.getValues(discordanceTrack, 0, 0))) {
+            return;
+        }
+
+        if (concordanceOnly && !isConcordant(vc, featureContext.getValues(concordanceTrack, 0, 0))) {
```

Comment:
Here too -- call the one-arg overload of `getValues()`: `getValues(final FeatureInput<T> featureDescriptor)`